### ISSUE TITLE
Schema of the Underlying data files

### ISSUE THREAD

**Issue by fivetran-tusharkumar** [apache/iceberg#6858]
### Question

Does iceberg track the schema of the underlying data file(eg. Parquet Files) which is referenced in the metadata? We know that the schema of the table is tracked, however is it possible to track the schema of the underlying files?

**Comment by Fokko** [comment]
Hey @fivetran-tusharkumar thanks for reaching out.

Iceberg is designed to do lazy changes. So if you add a column, this will be added to the table schema, but not to all the files. Once you read the files, the new column (that is missing from the Parquet file), will be added and this will be null. Once you rewrite a file, the file that replaces the file will have the new column. The Iceberg schema is optionally stored (some writers, do not write this unfortunately) in the [Parquet metadata](https://parquet.apache.org/docs/file-format/metadata/). Otherwise, you need to reconstruct the Iceberg schema from the Parquet schema that contains the FieldIDs that match with the Iceberg schema.

For example, in Iceberg if you rename a column. The old files that are already part of the table won't get rewritten right away but can be rewritten at some point eventually. Using the FieldIDs the columns are looked up, the old files are read with their original column name and then renamed to the new column name.

TLDR: You need to read the footer of each parquet file to determine the schema.
